model-optimizer/mo/graph/perm_inputs.py

```diff
@@ -132,14 +132,21 @@ def shape(op_node: Node, port_info: str, input_port: int):
     assert permutation_data_node.has_and_set('permutation'), 'Data node "{}" does not have permutation for node {}, ' \
                                                              'port_info "{}".'.format(permutation_data_node.id,
                                                                                       op_node.id, port_info)
-    permutation = permutation_data_node.permutation
-    if len(permutation.perm) == 0:
+    permute_idx_for_gather = permutation_data_node.permutation.perm
+    if len(permute_idx_for_gather) == 0:
         return
 
+    # Kludge, 'it ain't much honest, but it works' since StridedSlice need to be permuted even inputs
+    # have rank lesser than 4, e.g. input_shape = (1, 10, 10) input[..., newaxis]
+    if op_node.type == 'StridedSlice':
```

Comment:
`op_node.soft_get('type') == 'StridedSlice'`

Reply:
not present anymore